src/eureka/S3_data_reduction/s3_meta.py

```diff
@@ -274,7 +274,9 @@ def set_NIRISS_defaults(self):
         self.curvature = getattr(self, 'curvature', True)
         self.src_ypos = getattr(self, 'src_ypos', [35, 80])
         self.orders = getattr(self, 'orders', [1, 2])
+        self.all_orders = getattr(self, 'all_orders', [1, 2])
         self.record_ypos = getattr(self, 'record_ypos', False)
+        self.trace_offset = getattr(self, 'trace_offset', None)
```

Comment:
Please document all new ECF settings in the template ECFs and on the readthedocs pages

Reply:
It seems to me like we could add a subarray-specific default for `orders`, `all_orders`, and `trace_offset` here, rather than requiring user-input values or hard-coding defaults inside the `niriss.read` function (which makes tracking-down the default values tricky)

Reply:
Documentation added.

In exploring your 2nd point, I'm not sure how we would know what substrip mode is being used when calling `s3_meta.S3MetaClass()` when it's the first step in `s3_reduce.py`.  The ideal solution is to have PASTASOSS compute the trace correctly, at which point, we can get rid of the code in `niriss.get_wave()`.